frontend/src/libs/packages/map/map.package.ts

```diff
@@ -154,12 +181,12 @@ class MapService implements IMapService {
     position: google.maps.LatLngLiteral,
     isOrigin = false,
     angle = 0,
-  ): void {
+  ): google.maps.Marker {
     this.throwIfMapNotInitialized();
 
     const rotatedIconUrl = rotateImg(truckImg, angle);
```

Comment:
can you extract the logic of icon creation to helper?
it can be something like 
```
const createRotatedTruckIcon = (
  isOrigin: boolean,
  angle: number, 
  scale: number = DEFAULT_TRUCK_SCALE
) => {
   if (!isOrigin) {
      return truckImg;
   }
   const url = rotateImg(truckImg, angle);
   const anchor = new google.maps.Point(
              TRUCK_IMG_ANCHOR_X,
              TRUCK_IMG_ANCHOR_Y,
            );
   const size = new google.maps.Size(TRUCK_IMG_WIDTH, TRUCK_IMG_HEIGHT);
   return { url, anchor, size, scale }
}
```